tracing/http.go

```diff
@@ -20,9 +20,38 @@ import (
 	"net/http"
 
 	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/trace"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var (
+	// neverSample forcibly turns off sampling.
+	neverSample = trace.StartOptions{
+		Sampler: trace.NeverSample(),
+	}
+	// underlyingSampling uses the underlying sampling configuration (normally via the ConfigMap
+	// config-tracing).
+	underlyingSampling = trace.StartOptions{}
 )
 
 // HTTPSpanMiddleware is a http.Handler middleware to create spans for the HTTP endpoint
 func HTTPSpanMiddleware(next http.Handler) http.Handler {
 	return &ochttp.Handler{Handler: next}
 }
+
+// HTTPSpanMiddlewareIgnoringPaths is an http.Handler middleware to create spans for the HTTP
+// endpoint, not sampling any request whose path is in pathsToIgnore.
+func HTTPSpanMiddlewareIgnoringPaths(pathsToIgnore ...string) func(http.Handler) http.Handler {
```

Comment:
Why not merge this with the previous function, and make this a private method?

If not, it's particularly strange to have:

`HTTPSpanMiddleware` takes an `http.Handler` as an argument and returns another `http.Handler`.

`HTTPSpanMiddlewareIgnoringPaths` **returns a function** with the same signature as `HTTPSpanMiddleware` uses.

These names are really long -- can we shorten them to something like: `FilteredHttpTrace` (and `HttpTrace`)?

Reply:
I need to match the [Middleware type](https://github.com/cloudevents/sdk-go/blob/3eb440c338c784a56b16c9785d872fe8b0ed379f/pkg/cloudevents/transport/http/options.go#L225) from the CloudEvents SDK:
```golang
type Middleware func(next nethttp.Handler) nethttp.Handler
```

So, I need to return a function. Or I suppose I could make a struct and the function just hangs off the struct?

I could combine the two functions by simply removing the existing `HTTPSpanMiddleware` and renaming the new function to the old name. The downside is that we will have code in [serving](https://github.com/knative/serving/blob/24ae126893d0b26ba621b5d874ed76d8d67cfcd3/cmd/activator/main.go#L235) that looks like:
```golang
ah = tracing.HTTPSpanMiddleware()(ah)
```

What is your preference?

Reply:
Discussed this in person; we settled on making `HTTPSpanMiddleware` be an instance of `HTTPSpanIgnoringPaths()`.

Reply:
Talked in person and made the requested changes.